design/one-pager-release-process.md

```diff
@@ -0,0 +1,321 @@
+  attention.
+* **Demo party**: This is a chance to celebrate and show off the fruits of our recent labors to the
+  rest of the team.  Each developer can show a demo of new things they have been working on to
+  collect feedback and disseminate information about our progress and limitations to the rest of the
+  team.
+* **Retrospective**: After a milestone is completed, a retrospective look back over it can be very
+  helpful at identifying where we were successful and what we need to improve on.  This retro should
+  result in a series of action items that will help us improve our processes.
+
+In an attempt to include all of these events in a **monthly** release timeline, we propose the below
+regularly recurring schedule for each month. Given our current trajectories, the actual release
+itself will occur mid-month, e.g. around the **15th of the month**.
+
+Note that every month will have different alignment for weekdays and weekends, so this schedule is a
+guide to the overall structure, not a set of specific dates that will be used every month. The team
+should define the specific days/events of each monthly schedule **before** its cycle starts and
+those dates should be published to a new `release-schedule.md` that is linked from the main readme
+and roadmap.
+
+| Day of release cycle | Event | Expectations |
+| ------------ | ----- | ------------ |
+|**Week 1**|||
+|1 Mon|Milestone starts|(no actual event or meeting held for this)|
+|1 Mon|Weekly planning|Group planning session to identify priorities and focus for the week.|
+|3 Wed|Backlog grooming & epic discovery|(optional) Session to triage new issues and help investigate and flesh out epics.  Early in the milestone, the emphasis is more on epic discovery since we are still working through understanding the epics.|
+|4 Thurs|Retrospective from previous release|A public formal retrospective is held for the community to participate in. The events of the previous milestone will be discussed to identify improvements.|
+|**Week 2**|||
+|8 Mon|Weekly planning|Group planning session to identify priorities and focus for the week.|
+|10 Wed|Backlog grooming|(optional) Session to triage new issues and prioritize the backlog.|
+|12 Fri|Halftime Demo party|Show off our progress in demos for the team and take in critical feedback and discuss potential course corrections to set effective strategy.|
+|**Week 3**|||
+|15 Mon|Weekly planning|Group planning session to identify priorities and focus for the week.|
+|17 Wed|Backlog grooming|(optional) Session to triage new issues and prioritize the backlog.|
+|**Week 4**|||
+|22 Mon|Weekly planning|Group planning session to identify priorities and focus for the week.|
+|25 Wed|Backlog grooming & Milestone planning|Session to triage new issues as we get closer to feature freeze.  Milestone planning begins for the **next** milestone as we identify themes, priorities, and high level epics that we want to address in the **next** release.|
```

Comment:
Based on how some previous retrospectives have gone, I could see it being beneficial to hold the retrospective before planning, so that feedback from the retrospective can be incorporated into planning earlier. Otherwise, changes would need to wait until the following planning cycle, which is three weeks after the retrospective, and it may be hard to remember everything we wanted to do. How do we imagine that the process as written would incorporate feedback from the retrospectives?

That said, I am assuming we'll be adjusting the process over time, so I'm not sure this comment needs to block merging the document.

My proposal would be that if we did the retrospective on day 1, and also started with milestone planning on day 1 (with the intent that we'd finish it by the end of week 1), then it may be easier to incorporate the ideas from the retrospective into the planning. There could also be an additional benefit of allowing people to focus on the planning instead of closing the release. One drawback would be that there would be a reset period in between actual work on the milestones. What do other people think?

Reply:
I can definitely understand the the benefits of having the retrospective so that informs the planning.  However, I'm struggling with how we'd be able to fit in the retrospective (either per SIG or joint one for the entire community) and run planning sessions (per SIG) all within the same day and also accommodate all time zones.

Maybe we could do the retrospective on day 1 and then day 2 has planning?  This schedule will be flexible and open to changes over time as we experience it and learn from it, so it's not entirely set in stone.